internal/k8shandler/collection.go

```diff
@@ -199,17 +206,27 @@ func (clusterRequest *ClusterLoggingRequest) addSecurityLabelsToNamespace() erro
 	return nil
 }
 
-func (clusterRequest *ClusterLoggingRequest) RemoveInputServices() error {
+func (clusterRequest *ClusterLoggingRequest) RemoveInputServices(httpInputs []string, currOwner []metav1.OwnerReference, removeAllServices bool) error {
 	if clusterRequest.Cluster.Spec.Collection.Type != logging.LogCollectionTypeVector {
 		return nil
 	}
 
-	for _, input := range clusterRequest.Forwarder.Spec.Inputs {
-		if input.Receiver != nil && input.Receiver.HTTP != nil {
-			if err := clusterRequest.RemoveInputService(input.Name); err != nil {
+	// Find all http input services by label
+	labelSelector, _ := labels.Parse("component=http-input-service")
+	httpServices := corev1.ServiceList{}
+	if err := clusterRequest.Client.List(context.TODO(), &httpServices, &client.ListOptions{LabelSelector: labelSelector}); err != nil {
```

Comment:
Not sure if it's possible, but can we include the owner reference into the query so we only get services owned by currOwner and with "component=http-input-service" label?

Reply:
Unfortunately it is not possible to query on `OwnerRefs`